homeassistant/components/meater/sensor.py

```diff
@@ -0,0 +1,173 @@
+    """Set up the Meater Temperature Probe sensor."""
+    return True
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the entry."""
+    # assuming API object stored here by __init__.py
+    api = hass.data[DOMAIN][entry.entry_id]
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                devices = await api.get_all_devices()
+        except Exception as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        # Populate the entities
+        entities = []
+
+        for dev in devices:
+            if dev.id not in hass.data[DOMAIN]["entities"]:
+                entities.append(
+                    MeaterProbeTemperature(
+                        coordinator, dev.id, TemperatureMeasurement.Internal
+                    )
+                )
+                entities.append(
+                    MeaterProbeTemperature(
+                        coordinator, dev.id, TemperatureMeasurement.Ambient
+                    )
+                )
+                device_registry = await dr.async_get_registry(hass)
+                device_registry.async_get_or_create(
+                    config_entry_id=entry.entry_id,
+                    identifiers={(DOMAIN, dev.id)},
+                    name="Meater Probe " + dev.id,
+                    manufacturer="Apption Labs",
+                    model="Meater Probe",
+                )
+                hass.data[DOMAIN]["entities"][dev.id] = None
+
+        async_add_entities(entities)
+
+        return devices
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="meater_api",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=30),
+    )
+
+    def null_callback():
+        return
+
+    # Add a subscriber to the coordinator that doesn't actually do anything, just so that it still updates when all probes are switched off
+    coordinator.async_add_listener(null_callback)
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+
+class MeaterProbeTemperature(CoordinatorEntity):
+    """Meater Temperature Sensor Entity."""
+
+    def __init__(self, coordinator, device_id, temperature_reading_type):
+        """Initialise the sensor."""
+        self.coordinator = coordinator
+        self.device_id = device_id
+        self.temperature_reading_type = temperature_reading_type
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Meater Probe " + self.temperature_reading_type.name
```

Comment:
I believe literal string interpolation is the standard

Reply:
Resolved.